kitty: fix locating libstartup-notification-1

vlaci opened the pull request:
Startup notification doesn't work in recent versions of Kitty:
> Traceback (most recent call last):
>   File "/nix/store/3a3b0xd952gp8jw70k5kh3a4zhgzf0p7-kitty-0.12.3/bin/../lib/kitty/kitty/utils.py", line 216, in init_startup_notification
>     return init_startup_notification_x11(window_handle, startup_id)
>   File "/nix/store/3a3b0xd952gp8jw70k5kh3a4zhgzf0p7-kitty-0.12.3/bin/../lib/kitty/kitty/utils.py", line 201, in init_startup_notification_x11
>     return init_x11_startup_notification(display, window_handle, sid)
> OSError: Failed to load libstartup-notification-1.so with error: libstartup-notification-1.so: cannot open shared object file: No such file or directory

###### Motivation for this change

Apparently dispatching of startup notification has been moved to a C
binding in kitty 0.6.0 [1] so the substituion had to be modified to reflect
that. Without this fix Kitty still works except that window managers
which depend on startup notifications to be fired (e.g. Awesome)
cannot apply special placement rules.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

\[1\]: https://github.com/kovidgoyal/kitty/commit/b08f4ab5937199187581aa20e3a7aba7b2a34ac6

Commit:
kitty: fix locating libstartup-notification-1

Startup notification doesn't work in recent versions of Kitty:
> Traceback (most recent call last):
>   File "/nix/store/3a3b0xd952gp8jw70k5kh3a4zhgzf0p7-kitty-0.12.3/bin/../lib/kitty/kitty/utils.py", line 216, in init_startup_notification
>     return init_startup_notification_x11(window_handle, startup_id)
>   File "/nix/store/3a3b0xd952gp8jw70k5kh3a4zhgzf0p7-kitty-0.12.3/bin/../lib/kitty/kitty/utils.py", line 201, in init_startup_notification_x11
>     return init_x11_startup_notification(display, window_handle, sid)
> OSError: Failed to load libstartup-notification-1.so with error: libstartup-notification-1.so: cannot open shared object file: No such file or directory

Apparently dispatching of startup notification has been moved to a C
binding in kitty 0.6.0 [1] so the substituion had to be modified to reflect
that. Without this fix Kitty still works except that window managers
which depend on startup notifications to be fired (e.g. Awesome)
cannot apply special placement rules.

The substitution mechanism is also changed to use a patch file to detect
this kind of mistakes in the future.

[1]: https://github.com/kovidgoyal/kitty/commit/b08f4ab5937199187581aa20e3a7aba7b2a34ac6

jtojnar commented on pkgs/applications/misc/kitty/default.nix:
```diff
@@ -29,8 +29,8 @@ buildPythonApplication rec {
   outputs = [ "out" "terminfo" ];
 
   postPatch = ''
-    substituteInPlace kitty/utils.py \
```
Could you move this into a patch file using `substituteAll`? Then issue like this will not happen in the future.

Mic92 replied:
Example: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/video/gnome-mplayer/default.nix#L20

vlaci replied:
Sure, will do in an additional commit and include the other replacement in it too.

vlaci replied:
Hm, the other substitution in the Makefile is f[ishy too](https://hydra.nixos.org/build/82946154/log):
```PYTHONPATH=YTHONPATH:.. HOME=MPDIR/nowhere sphinx-build -M html "." "_build" -T  ```
The variable substituioons without curly brackets are invalid in makefiles. Will fix it too.

Mic92 commented:
Does this also affect 18.09?

vlaci commented:
Yes it does, however the Pyhon backtrace is not shown when using the 18.09 version which is `0.11.3` if I am not mistaken.

Commit:
kitty: fixing environment variable substituition in docs/Makefile

The previous substitutution only worked by accident as make requires
environment variables to be enclosed in curly brackets as shown in this
excerpt from the build output:
> PYTHONPATH=YTHONPATH:.. HOME=MPDIR/nowhere sphinx-build -M html "." "_build" -T

The substituition is moved to the existing patch file to simplify the
builder expression.

Mic92 commented:
backport: 4d9183342bb4c833895a6f791f3be003fce6ab6c
